### PR TITLE
modify doc's example of tf.print to wotk TF 2.0

### LINKED ISSUE #27478: tf.print the example doesn't work with TF 2.0
<em>Please make sure that this is a documentation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:doc_template</em>

**System information**
- TensorFlow version: 2.0
- Doc Link: https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/print

**Describe the documentation issue**
the example doesn't work with TF 2.0
```python
    tf.enable_eager_execution()
    @tf.contrib.eager.defun
    def f():
        tensor = tf.range(10)
        tf.print(tensor, output_stream=sys.stderr)
        return tensor
    range_tensor = f()
```
```python
    sess = tf.Session()
    with sess.as_default():
        tensor = tf.range(10)
        print_op = tf.print("tensors:", tensor, {2: tensor * 2},
                            output_stream=sys.stdout)
        with tf.control_dependencies([print_op]):
          tripled_tensor = tensor * 3
        sess.run(tripled_tensor)
```
**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

### PR DESCRIPTION
**Pull request by hksonngan**
fixed #27478

### CODE REVIEW

File: tensorflow/python/ops/logging_ops.py [hunk 1]
```diff
@@ -72,13 +72,10 @@
...
-    sess = tf.Session()
-    with sess.as_default():
-        tensor = tf.range(10)
-        print_op = tf.print(tensor)
-        with tf.control_dependencies([print_op]):
-          out = tf.add(tensor, tensor)
-        sess.run(out)
+    tensor = tf.range(10)
+    print_op = tf.print(tensor)
+    with tf.control_dependencies([print_op]):
```
Comment (alextp) [inline tensorflow/python/ops/logging_ops.py]: No control dependencies in tfv2

File: tensorflow/python/ops/logging_ops.py [hunk 2]
```diff
@@ -176,9 +171,8 @@ def print_v2(*inputs, **kwargs):
 
     Usage in a defun:
     ```python
-    tf.enable_eager_execution()
-
-    @tf.contrib.eager.defun
+    from tensorflow.python.eager import function
```
Comment (alextp) [inline tensorflow/python/ops/logging_ops.py]: tf.function exists so no need to import from contrib which doesn't exist in tf2

File: tensorflow/python/ops/logging_ops.py [hunk 3]
```diff
@@ -72,13 +72,10 @@
...
                           "of how to ensure tf.print executes in graph mode:\n"
                           """```python
-    sess = tf.Session()
-    with sess.as_default():
-        tensor = tf.range(10)
-        print_op = tf.print(tensor)
-        with tf.control_dependencies([print_op]):
-          out = tf.add(tensor, tensor)
-        sess.run(out)
+    tensor = tf.range(10)
```
Comment (allenlavoie) [inline tensorflow/python/ops/logging_ops.py]: The deprecation message I think is fine using a tf.Session; it's giving directions to 1.x users about how to switch within 1.x.

File: tensorflow/python/ops/logging_ops.py [hunk 4]
```diff
@@ -190,14 +184,11 @@ def f():
 
     Usage when constructing graphs:
```
Comment (allenlavoie) [inline tensorflow/python/ops/logging_ops.py]: The best thing to do with 1.x vs. 2.x docstrings is to make a new symbol which is v2-only with the new docstring (no need to mention control dependencies at all there, and certainly no need to mention Sessions) and have a v1-only symbol with the same name which has the old docstring. The v1 symbol should call the v2 symbol.

We're still planning on making a couple 1.x releases I believe, so we should preserve the existing information in the v1 docstring.

Example of this kind of transformation (although it's copying a bit more than you'll need to): https://github.com/tensorflow/tensorflow/commit/dd0b3d3d724f13bb99423e92ea3e32601058ff2a

Note the tf_export lines, which control which versions the symbol is exposed in.

**Review by allenlavoie — COMMENTED** [review]
Thanks for looking into this!

File: tensorflow/python/ops/logging_ops.py (hunk 4 above)
Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]: @allenlavoie I finished the work as [`dd0b3d3`](https://github.com/tensorflow/tensorflow/commit/dd0b3d3d724f13bb99423e92ea3e32601058ff2a), but do you thing is good to double the code is just for documenting? Or should I leave or close this issue, and re-open when release when finalizing tfv2?

Comment (allenlavoie) [reply tensorflow/python/ops/logging_ops.py]: A little duplication is fine. They're giving different information.

File: tensorflow/python/ops/logging_ops.py (hunk 1 above)
Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]: this is document [control_dependencies](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/control_dependencies), also this code works with tfv2 nightly build.

File: tensorflow/python/ops/logging_ops.py (hunk 3 above)
Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]: add comment as [`b38ef23`](https://github.com/tensorflow/tensorflow/pull/27479/commits/b38ef235d56a340cb080da328f47294ce24a92ec)

File: tensorflow/python/ops/logging_ops.py (hunk 4 above)
Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]: @allenlavoie  I modify as [`b38ef23`](https://github.com/tensorflow/tensorflow/pull/27479/commits/b38ef235d56a340cb080da328f47294ce24a92ec)

File: tensorflow/python/ops/logging_ops.py (hunk 2 above)
Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]: `from tensorflow.python.eager import function` is for decorator function.defun.
refer to [`control flow unit test`](https://github.com/tensorflow/tensorflow/blob/1d54cbf4a2252215c5d2ce9accb5e498a7c2a704/tensorflow/python/kernel_tests/control_flow_util_v2_test.py)

Comment (allenlavoie) [reply tensorflow/python/ops/logging_ops.py]: We need to use the public API in our documentation.

Comment (hksonngan) [reply tensorflow/python/ops/logging_ops.py]:
```python
from tensorflow.python.framework import function
@function.Defun()
def f():
  tensor = tf.range(10)
  tf.print(tensor, output_stream=sys.stderr)
  return tensor

range_tensor = f()
```
raise InvalidArgumentError as is expected

**Comment by rthadur** [comment]
@hksonngan can you please resolve conflicts

**Comment by hksonngan** [comment]
@rthadur done.

File: tensorflow/python/ops/logging_ops.py [hunk 5]
```diff
@@ -63,16 +63,15 @@
...
-                          "this operator will not be executed unless it is "
-                          "directly specified in session.run or used as a "
-                          "control dependency for other operators. This is "
-                          "only a concern in graph mode. Below is an example "
-                          "of how to ensure tf.print executes in graph mode:\n"
-                          """```python
-    # tf.Session is deprecated in TF 2
-    sess = tf.Session()
+            "tf.print returns a no-output operator that directly "
+            "prints the output. Outside of defuns or eager mode, "
```
Comment (alextp) [inline tensorflow/python/ops/logging_ops.py]: s/defuns/tf.function

File: tensorflow/python/ops/logging_ops.py [hunk 6]
```diff
@@ -63,16 +63,15 @@
...
+            "this operator will not be executed unless it is "
+            "directly specified in session.run or used as a "
+            "control dependency for other operators. This is "
+            "only a concern in graph mode. Below is an example "
```
Comment (alextp) [inline tensorflow/python/ops/logging_ops.py]: only a concern in legacy v1 graph mode

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/logging_ops.py (hunk 1 above)
Comment (alextp) [reply tensorflow/python/ops/logging_ops.py]: Yes but you don't need the control dependency if inside a tf.function or with eager execution enabled

File: tensorflow/python/ops/logging_ops.py (hunk 2 above)
Comment (alextp) [reply tensorflow/python/ops/logging_ops.py]: Yes but don't use the deprecated function.Defun in documentation

File: tensorflow/python/ops/logging_ops.py [hunk 7]
```diff
@@ -127,15 +127,15 @@ def _is_filepath(output_stream):
   """Returns True if output_stream is a file path."""
   return isinstance(output_stream, str) and output_stream.startswith("file://")
 
-
+# todo: remove this when release tfv2
```
Comment (alextp) [inline tensorflow/python/ops/logging_ops.py]: Remove the TODO

**Review by alextp — APPROVED** [review]

**Comment by allenlavoie** [comment]
This PR is in the process of being rolled back since a broke some tests. Could you return the output of `print_v2` (should be an `Operation`) rather than just calling it? Then we can re-submit.

**Comment by hksonngan** [comment]
@allenlavoie like this [4a5cfae](https://github.com/hksonngan/tensorflow/commit/4a5cfae6017c2b4f6e541b400d513e5fffda9265)?

**Comment by allenlavoie** [comment]
Yes, that looks right to me.

**Pull request merged**